doc/dev/static_type_checking.md

```diff
@@ -0,0 +1,1452 @@
+
+To run pyright on your library, run the tox pyright env at the package level:
+
+`.../azure-sdk-for-python/sdk/textanalytics/azure-ai-textanalytics>tox -e pyright -c ../../../eng/tox/tox.ini`
+
+If you don't want to use `tox` you can also install and run pyright on its own:
+
+`pip install pyright==TODO`
+
+`.../azure-sdk-for-python/sdk/textanalytics/azure-ai-textanalytics>pyright azure`
+
+Note that you may see different errors if running a different version of pyright than the one in CI.
+
+Configuration of pyright is handled globally for the repo, but if specific configuration of pyright is needed for your library, use a `pyrightconfig.json` file at the package level.
+
+For example, to ignore type checking files under a specific directory, you can use `exclude`:
+
+```json
+{
+  "exclude": ["**/_dont_type_check_me/**"]
+}
+```
+
+Full documentation on pyright config options can be found here: https://github.com/microsoft/pyright/blob/main/docs/configuration.md
+
+### How to ignore type checking errors
+
+A type checking error can be globally ignored by placing a `# type: ignore` comment on the offending line. Generally, we do not want to
+ignore errors if we can, but sometimes it is not possible to fix the error, e.g. there might be bug in the type checker _or_
+the Python code is more expressive than the type system allows. If you must ignore an error, it is
+recommended that you explain your reasoning in the ignore comment and include a link to the bug or Github issue so someone
+reading the code later may resolve it.
+
+*Globally ignore (both mypy and pyright):*
+```python
+ignore_me: int = 5  # type: ignore  https://www.github.com/Azure/azure-sdk-for-python/issues/my-issue
+```
+
+Note that it is possible to be specific in the error to ignore, instead of globally turning off type checking on
+that line. Syntax for this involves putting the specific error code in brackets of the ignore comment. Error codes
+are found next to the type checking error. 
+
+*mypy:*
+```python
+ignore_me: int = 5  # type: ignore[misc]
+```
+
+*pyright:*
+```python
+ignore_me: int = 5  # pyright: ignore[reportPrivateUsage]
+```
+
+The type checkers will continue to evaluate the line of code for other typing errors.
+
+
+More information about suppressing errors can be found in the
+official [mypy docs](https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html?highlight=ignore#silencing-type-errors) and [pyright docs](https://github.com/microsoft/pyright/blob/main/docs/comments.md#line-level-diagnostic-suppression).
+
+
+### How to opt out of type checking
+
+All client libraries in the Python SDK repo are automatically opted in to running type checking. If there is a
+reason why a particular library should not run type checking, it is possible to add that library to a block list to prevent mypy/pyright
+from running checks.
+
+1) Place the package name on this block list: TODO
```

Comment:
It will be a new file, e.g. type_checker_ignore_list or something. You will need to add your package to the file if you want to _opt-out_ of type checking in CI.